corelib-search/src/main/java/eu/europeana/corelib/search/impl/WebMetaInfo.java

```diff
@@ -261,4 +258,36 @@ public static boolean hasReferencedBy(FullBean bean) {
         LOG.debug("hasReferencedBy = false");
         return false;
     }
-}
+
+    /**
+     * Check if the dctermsIsReferencedBy is not null then add the preview instances for IIIF. check ticket: EA-1184
+     * @param bean
+     * @param manifestAddUrl
+     * @return
+     */
+    private static void addPreviewInstancesIIIF(FullBean bean, String manifestAddUrl) {
+        long start = System.nanoTime();
+        //check if manifestAddUrl property is present and true
+        if (manifestAddUrl != null && manifestAddUrl.equals("true")) {
+            for (Aggregation a : bean.getAggregations()) {
+                // add to all webresources in all aggregations
+                for (WebResource wr : a.getWebResources()) {
+                    //check if dctermsIsReferencedBy is not null or empty
+                    if (wr.getDctermsIsReferencedBy() != null && wr.getDctermsIsReferencedBy().length > 0) {
+                        String result[]= new String[wr.getDctermsIsReferencedBy().length];
+                        int i=0;
+                        for (String value : wr.getDctermsIsReferencedBy()) {
+                            result[i]=value + MANIFEST_PREVIEW_INSTANCE;
+                            i++;
+                        }
+                        wr.setDctermsIsReferencedBy(result);
+                    }
+                }
+            }
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("AddPreviewInstancesIIIF took {} ns", System.nanoTime() - start);
+            }
```

Comment:
We already loop over all aggregations and webresources twice in the code, and this is adding a 3rd time. I think we should refactor and try to reduce the iterations to 2 times. 2 times is the minimum because before we can start adding data we have to check all webresources to see if there is any with a dcTermsHasReferenced field.
I talked to Hugo and luckily he's fine with us only adding the extra part when we construct a manifest url here, so we can simply remove this addPreviewInstances method and move the check+adding extra part to the addReferencedByIIIF method